src/main/java/hudson/remoting/forward/CopyThread.java

```diff
@@ -20,6 +20,9 @@ public CopyThread(String threadName, InputStream in, OutputStream out) {
         super(threadName);
         this.in = in;
         this.out = out;
+        setUncaughtExceptionHandler((t, e) -> {
+            LOGGER.log(Level.WARNING, "Uncaught exception in CopyThread " + t, e);
+        });
```

Comment:
Ideally creators of this thread should pass additional termination handler:

* PortForwarder - do handling and close the socket ?
* ForwarderFactory#connect - Interrupt RemoteOutputStream ?